packages/keyring-controller/tests/mocks/mockEncryptor.ts

```diff
@@ -35,13 +34,13 @@ export default class MockEncryptor implements ExportableKeyEncryptor {
       throw new Error(INVALID_PASSWORD_ERROR);
     }
 
-    return cacheVal ?? {};
+    return JSON.parse(cacheVal) ?? {};
```

Comment:
This is due to an interesting discovery regarding the QRKeyring. Apparently, the `serialize` method of the keyring does not return an object compatible with what we internally call "serializable" (i.e. `Json` type), as it contains keys with `undefined` as value 

The only reason why this does not break production is that the `encryptor` calls `JSON.stringify`, which removes any key with undefined as value.

Without this, the unlock operation would be broken after adding the QRKeyring, as the `isSerializableKeyringArray` assertion would fail

Reply:
Due to this, I'm also realising that we make sure that what is unlocked from the vault is a correct serializable object, but we don't do that when locking and saving the keyrings _to the vault_ (the other way around): this means that we are not sure that we are saving a proper object in the vault, but we expect it to be when reading from it.

Perhaps we could add the same check in `persistAllKeyrings`, even if it is not required type-wise. This would ensure that each keyrings implement `serialize` properly (given that this is our requirement)